### DISCUSSION TITLE
What are the best practices for managing version control of translation keys in a multilingual project using Tolgee?

### DISCUSSION THREAD

**Discussion by aryanarayantiwari** [tolgee/tolgee-platform#2451] [General]
How can I effectively manage version control for translation keys when working on a multilingual project using Tolgee? I want to ensure that translations remain consistent across different app versions and that any changes or additions to translation keys are tracked and managed efficiently. What are the best practices for keeping translation keys organized, synchronized with the codebase, and compatible with various environments (e.g., development, staging, production)? Additionally, how can I ensure that old or deprecated keys are handled properly?

**Comment by stepan662** [comment]
Hey, I recommend reading this article: https://tolgee.io/blog/organize-your-translation-keys-with-tags

**Reply by aryanarayantiwari** [reply]
I have already reviewed the article, but it doesn't meet my requirements. I want to manage translation keys by tagging them in a way that supports multiple versions. For example, I have an application with versions 1.0 and 1.1, and I need to tag the translation keys so I can maintain different values for each version and easily revert to a previous version if necessary.

**Reply by JanCizmar** [reply]
Hello! This looks like a request for a branching feature, which we indeed have on the roadmap. However, we have been talking to multiple companies about it, and it doesn't seem like the best way to do the job because it will add new complexity and problems similar to versioning system problems like Git. We don't want to reimplement Git, so we made it simple for now.

So how do we solve it?
1. When you need to dramatically change a string, change the key. Dramatic changes between values are usually a sign that you might be willing to change the key. 
2. Another option: If you don't need to make heavy changes to the older versions, you can stop syncing them with the Tolgee platform and fix the necessary stuff directly in the exported files.
3. Another approach would be creating a separate Tolgee project for an older version.
